### DISCUSSION TITLE
Search by number of elements

### DISCUSSION THREAD

**Discussion by xxxserxxx** [TomWright/dasel#116] [Ideas]
Is there a search syntax that tests for the cardinality of a child array?  Something like:

```
echo '{"item1": {"name": "foo", "children": []}, "item2": {"name": "bar", "children": [1,2]} | \
dasel '.(?: #.children > 0)'
```

would return the `item2` object.

**Comment by TomWright** [comment]
This isn't a feature at the moment but I'd be happy to include it in the future.

I'll do some thinking and plan this out soon.

**Comment by xxxserxxx** [comment]
Hey, sorry -- I forgot to put REQ: in front of the subject. I was in a rush, and was sloppy.

**Reply by TomWright** [reply]
No worries at all!

I'm thinking of a syntax to select the length of an element so as it's part of the standard selector syntax - any suggestions?

**Comment by xxxserxxx** [comment]
Not really. I don't yet have a feel for the style of Dasel.  Current array selectors are like `.[5]` for the 6th element, and `.[]` for the next index, so I'm inclined to suggest something with `#` -- probably because pound is so common in other languages. Maybe `.[#]` or `#.[]`.

**Reply by TomWright** [reply]
I like the look of `.[#]`, and will be able to re-use to select the length of maps and strings too.

I think I'll add that selector + the following comparisons when searching:
- `>`
- `>=`
- `<`
- `<=`

**Comment by xxxserxxx** [comment]
That would do it!

**Comment by TomWright** [comment]
@xxxserxxx This is now supported as of `v1.14.0`. I'll be releasing it shortly.

```
echo '{
  "item1": {
    "name": "foo",
    "children": []
  },
  "item2": {
    "name": "bar",
    "children": [
      1,
      2
    ]
  }
}' | dasel -p json -m '.(?:.children.[#]>0).name'

"bar"
```

Related docs are here:
- https://daseldocs.tomwright.me/selectors/length
- https://daseldocs.tomwright.me/selectors/dynamic

**Comment by xxxserxxx** [comment]
That's beautiful, thanks!
